lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -27,9 +27,118 @@ class AwsCompileCloudFrontEvents {
+        Id: { type: 'string' },
+        OriginCustomHeaders: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              HeaderName: { type: 'string' },
+              HeaderValue: { type: 'string' },
+            },
+            additionalProperties: false,
+            required: ['HeaderName', 'HeaderValue'],
+          },
+        },
+        OriginPath: { type: 'string' },
+        S3OriginConfig: {
+          type: 'object',
+          properties: {
+            OriginAccessIdentity: {
+              type: 'string',
+              pattern: '^origin-access-identity/cloudfront/.+',
+            },
+          },
+          additionalProperties: false,
+        },
+      },
+      additionalProperties: false,
+      required: ['DomainName', 'Id'],
+      oneOf: [{ required: ['CustomOriginConfig'] }, { required: ['S3OriginConfig'] }],
+    };
+
+    const behaviorObjectSchema = {
+      type: 'object',
+      properties: {
+        AllowedMethods: {
+          oneOf: [
+            { type: 'array', items: [{ const: 'GET' }, { const: 'HEAD' }] },
+            { type: 'array', items: [{ const: 'GET' }, { const: 'HEAD' }, { const: 'OPTIONS' }] },
+            {
+              type: 'array',
+              items: [
+                { const: 'GET' },
+                { const: 'HEAD' },
+                { const: 'OPTIONS' },
+                { const: 'PUT' },
+                { const: 'PATCH' },
+                { const: 'POST' },
+                { const: 'DELETE' },
+              ],
+            },
+          ],
+        },
+        CachedMethods: {
+          oneOf: [
+            { type: 'array', items: [{ const: 'GET' }, { const: 'HEAD' }] },
+            { type: 'array', items: [{ const: 'GET' }, { const: 'HEAD' }, { const: 'OPTIONS' }] },
+          ],
+        },
+        CachePolicyId: { type: 'string' },
+        Compress: { type: 'boolean' },
+        FieldLevelEncryptionId: { type: 'string' },
+        OriginRequestPolicyId: { type: 'string' },
+        SmoothStreaming: { type: 'boolean' },
+        TrustedSigners: { type: 'array', items: { type: 'string' } },
+        ViewerProtocolPolicy: {
+          type: 'string',
+          enum: ['allow-all', 'redirect-to-https', 'https-only'],
+        },
+      },
+      additionalProperties: false,
+    };
+
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'cloudFront', {
       type: 'object',
+      properties: {
+        eventType: {
+          type: 'string',
+          enum: ['viewer-request', 'origin-request', 'origin-response', 'viewer-response'],
+        },
+        pathPattern: { type: 'string', pattern: '^([A-Za-z0-9_.*$/~"\'@:+-]|&amp;)+$' },
+        origin: {
+          oneOf: [{ type: 'string', format: 'uri' }, originObjectSchema],
+        },
+        isDefaultOrigin: { type: 'boolean' },
+        includeBody: { type: 'boolean' },
+        behavior: behaviorObjectSchema,
```

Comment:
Let's ensure alphabetical order (when there's larger count of properties, it's then easier to find one)